gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/ContentsUtil.java

```diff
@@ -57,29 +58,37 @@ public static Optional<String> fetchPageContents(URL pageUrl) {
 
     private static Callable<String> fetchContentTask(URL pageUrl, ProgressIndicator indicator) {
         return () -> {
-            HttpURLConnection connection = (HttpURLConnection) pageUrl.openConnection();
-            connection.setRequestMethod("GET");
+            HttpURLConnection connection = null;
+            try {
+                connection = (HttpURLConnection) pageUrl.openConnection();
+                connection.setRequestMethod("GET");
 
-            if (connection.getResponseCode() != 200) {
-                connection.disconnect();
-                throw new ProcessCanceledException();
-            }
+                if (connection.getResponseCode() != 200) {
+                    throw new ProcessCanceledException();
+                }
 
-            BufferedReader in =
-                    new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
-            StringBuilder result = new StringBuilder();
-            String inputLine;
+                BufferedReader in =
+                        new BufferedReader(new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+                StringBuilder result = new StringBuilder();
+                String inputLine;
 
-            while ((inputLine = in.readLine()) != null) {
-                if (indicator.isCanceled()) {
-                    throw new InterruptedException("Fetch cancelled");
+                while ((inputLine = in.readLine()) != null) {
+                    if (indicator.isCanceled()) {
+                        throw new InterruptedException("Fetch cancelled");
+                    }
+                    result.append(inputLine);
```

Comment:
this seems somewhat verbose, could you not just do:

```java
return in.lines().collect(Collectors.joining('\n'));
```

?

Also, I think if the underlying content contains new lines, your current implementation discards them.